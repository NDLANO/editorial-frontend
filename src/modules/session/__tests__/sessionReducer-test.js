/**
 * Copyright (c) 2016-present, NDLA.
 *
 * This source code is licensed under the GPLv3 license found in the
 * LICENSE file in the root directory of this source tree.
 *
 */

import reducer, { getSessionStateFromLocalStorage, actions } from '../session';

test('reducers/session getSessionStateFromLocalStorage when id_token exists in local storage', () => {
  localStorage.setItem('id_token', 'eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJhcHBfbWV0YWRhdGEiOnsibmRsYV9pZCI6IlFhZHZwNjk5aUh0ak9tREhNQ2xtWGJBLSJ9LCJuYW1lIjoiw5h5dmluZCBNYXJ0aGluc2VuIiwiaXNzIjoiaHR0cHM6Ly9uZGxhLmV1LmF1dGgwLmNvbS8iLCJzdWIiOiJnb29nbGUtb2F1dGgyfDEwNzA2NDA0NTM4NTY0ODQwNDgxNiIsImF1ZCI6IldVMEtyNENEa3JNMHVMOXhZZUZWNGNsOUdhMXZCM0pZIiwiZXhwIjoxNDk2MjM2MDAzLCJpYXQiOjE0OTYyMzU0MDMsIm5vbmNlIjoibFE1TVJacFY2ODlubVpPQzA5QlNRaTE5Rn4xNWY5OUEiLCJhdF9oYXNoIjoiOUZCSXRSQ1h4ZlNTaG1rMEpEZUtJUSJ9.-LhugcAhXJltZ6KI9d1hzR8XklaDCFK7AsmQrZ72oXA9');
  const nextState = getSessionStateFromLocalStorage();

  expect(nextState).toMatchSnapshot();
});

test('reducers/session getSessionStateFromLocalStorage when id_token does not exists in local storage', () => {
  localStorage.removeItem('id_token');

  const nextState = getSessionStateFromLocalStorage();
  expect(nextState).toMatchSnapshot();

  localStorage.setItem('id_token', '12345678');
});

test('reducers/session initialState when id_token exists in localstorage', () => {
  const nextState = reducer(undefined, { type: 'Noop' });

  expect(nextState).toMatchSnapshot();
});

test('reducers/session sets if authenticated', () => {
  const nextState = reducer(undefined, { type: actions.setAuthenticated, payload: true });
  expect(nextState).toMatchSnapshot();
});


test('reducers/session sets if authenticated', () => {
  const nextState = reducer(undefined, { type: actions.setAuthenticated, payload: false });
  expect(nextState).toMatchSnapshot();
});

test('reducers/session sets user data', () => {
  const nextState = reducer(undefined, {
    type: actions.setUserData,
    payload: {
      name: 'Ola Nordmann',
    },
  });
  expect(nextState).toMatchSnapshot();
});

test('reducers/session clears user data', () => {
  const nextState = reducer({
    user: {
      name: 'Ola Nordmann',
    },
    authenicated: false,
  }, {
    type: actions.clearUserData,
  });

  expect(nextState.user).toEqual({});
});
