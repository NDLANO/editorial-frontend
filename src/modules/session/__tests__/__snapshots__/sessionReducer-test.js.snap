// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`reducers/session getSessionStateFromLocalStorage when id_token does not exists in local storage 1`] = `
Object {
  "authenticated": false,
  "user": Object {},
}
`;

exports[`reducers/session getSessionStateFromLocalStorage when id_token exists in local storage 1`] = `
Object {
  "authenticated": true,
  "user": Object {
    "app_metadata": Object {
      "ndla_id": "Qadvp699iHtjOmDHMClmXbA-",
    },
    "at_hash": "9FBItRCXxfSShmk0JDeKIQ",
    "aud": "WU0Kr4CDkrM0uL9xYeFV4cl9Ga1vB3JY",
    "exp": 1496236003,
    "iat": 1496235403,
    "iss": "https://ndla.eu.auth0.com/",
    "name": "Ã˜yvind Marthinsen",
    "nonce": "lQ5MRZpV689nmZOC09BSQi19F~15f99A",
    "sub": "google-oauth2|107064045385648404816",
  },
}
`;

exports[`reducers/session initialState when id_token exists in localstorage 1`] = `
Object {
  "authenticated": false,
  "user": Object {},
}
`;

exports[`reducers/session sets if authenticated 1`] = `
Object {
  "authenticated": true,
  "user": Object {},
}
`;

exports[`reducers/session sets if authenticated 2`] = `
Object {
  "authenticated": false,
  "user": Object {},
}
`;

exports[`reducers/session sets user data 1`] = `
Object {
  "authenticated": false,
  "user": Object {
    "name": "Ola Nordmann",
  },
}
`;
